### PR TITLE
Extract tenant and user id to request params constructor

### PR DESCRIPTION
**Pull request by lpichler**
- this moves functions getTenantFromEchoContext and getUserFromEchoContext to util package - so it could be user in other places
- add constructor (I am sure if there is other terminology ) ` NewRequestParamsFromContext` where we fetch tenant_id and user_id from echo context and it could be used in  get[model]DaoWithTenant function. In this PR it is done for  `getSourceDaoWithTenant` and `getAuthenticationDaoWithTenant`.

### Links
Depends on
- [x] https://github.com/RedHatInsights/sources-api-go/pull/424
- issue https://github.com/RedHatInsights/sources-api-go/issues/356

### CODE REVIEW

**Review by petracihalova — APPROVED** [review]
LGTM

File: util/context.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package util
```
Comment (lindgrenj6) [inline util/context.go]: [tiny]: can we rename this file `echo_context.go`? Context overlaps a bit with the standard library context package.

**Review by lindgrenj6 — COMMENTED** [review]
Just one tiny thing, otherwise lgtm.

File: util/context.go (hunk 1 above)
Comment (lpichler) [reply util/context.go]: sure, done, thanks 👍

**Pull request merged**